metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -3168,11 +3214,63 @@ def emit_site_container(self):
             sub_types=[c.SITE],
         )
 
+    def _fetch_groups(self):
+        for group in TSC.Pager(self.server.groups):
+            self.group_map[group.id] = group
+
+    def _filter_capabilities(self, capabilities):
```

Comment:
let's get some type annotations on this. I think this should be returning an `optional[list[str]]`, and the formatting stuff should be done by the caller